File: sqlbuilder/sqlbuilder/SingleTxManager.java
package com.clickability.cms.dataaccess.sqlbuilder;

import java.sql.SQLException;

/*
 * Use for insert/update/delete statements.
 * 
 * See Also: QueryManager.java for select statements.
 */
public class SingleTxManager extends UpdateManager implements TransactionManager {
	
	public SingleTxManager(String dbName) {
		super(dbName);
	}
	
	public void begin() throws SQLException {
		this.connection.setAutoCommit(false);
	}
	
	public void commit() throws SQLException {
		this.connection.commit();
	}

	public void rollback() throws SQLException {
		this.connection.rollback();
	}
	
	public void end() throws SQLException {
		this.connection.setAutoCommit(true);
		super.close();
	}
	
	@Override
	public void close() {
		// Don't autoclose the connection
		// it is needed for the rest of the transaction.
	}

}

File: queryBuilder/src/main/kotlin/queryBuilder/com/inunison/lib/queryBuilder/SingleTxManager.kt
package com.clickability.cms.dataaccess.sqlbuilder

import java.sql.SQLException

/*
 * Use for insert/update/delete statements.
 *
 * See Also: QueryManager.java for select statements.
 */
class SingleTxManager(dbName: String) : UpdateManager(dbName), TransactionManager {

    @Throws(SQLException::class)
    fun begin() {
        this.connection.setAutoCommit(false)
    }

    @Throws(SQLException::class)
    fun commit() {
        this.connection.commit()
    }

    @Throws(SQLException::class)
    fun rollback() {
        this.connection.rollback()
    }

    @Throws(SQLException::class)
    fun end() {
        this.connection.setAutoCommit(true)
        super.close()
    }

    @Override
    fun close() {
        // Don't autoclose the connection
        // it is needed for the rest of the transaction.
    }

}
